src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -201,10 +201,14 @@ private static ISymbol MapSymbol(ISymbol symbol, ITypeSymbol type)
                 type = semanticModel.GetTypeInfo(bindableParent, cancellationToken).Type;
                 declaredSymbol = MapSymbol(semanticFacts.GetDeclaredSymbol(semanticModel, token, cancellationToken), type);
 
-                allSymbols = semanticModel.GetSymbolInfo(bindableParent, cancellationToken)
-                    .GetBestOrAllSymbols()
-                    .WhereAsArray(s => !s.Equals(declaredSymbol))
-                    .SelectAsArray(s => MapSymbol(s, type));
+                var skipSymbolInfoLookup = declaredSymbol.IsKind(SymbolKind.RangeVariable);
+                allSymbols = skipSymbolInfoLookup
+                    ? ImmutableArray<ISymbol>.Empty
+                    : semanticFacts
+                        .GetSymbolInfo(semanticModel, bindableParent, token, cancellationToken)
+                        .GetBestOrAllSymbols()
+                        .WhereAsArray(s => !s.Equals(declaredSymbol))
+                        .SelectAsArray(s => MapSymbol(s, type));
```

Comment:
the previous code was bad.  However, since you're going to end with SelectAsArray, doing a WhereAsArray is not actually desirable as it forces two full arrays to be created.